spec/mongo/collection_spec.rb

```diff
@@ -2000,6 +2000,51 @@ def generate
   end
 
   describe '#count_documents' do
+
+        context 'no argument provided' do
+
+          # no arguments
+          let(:view) do
+            described_class.new(authorized_collection, {}, {})
+          end 
+
+          before do
+            authorized_collection.delete_many
+          end
+
+          context 'empty collection' do
+            it 'returns true' do
+              expect(view.count_documents()).to eq(view.count_documents({}))
+            end
+          end
+
+          context 'nonempty collection' do
+            it 'returns false' do
+              documents = []
+              1.upto(10) do |index|
+                documents << { key: 'a', _id: "in#{index}" }
+              end
+
+              authorized_collection.insert_many(documents)
```

Comment:
This can be done using lets and a before block:

```
let(:documents) do
   documents = []
   1.upto(10) do |index|
      documents << { key: 'a', _id: "in#{index}" }
   end
end

before do
   authorized_collection.insert_many(documents)
end

it "description here" do
   ...
end
```

Reply:
Which is not to say that this doesn't work, just that `let` and `before` are more idiomatic for respec and also help you separate out the logic of the test itself vs setup logic.